jctools-core/src/main/java/org/jctools/queues/atomic/BaseLinkedAtomicQueue.java

```diff
@@ -53,6 +53,55 @@ protected final void spConsumerNode(LinkedQueueAtomicNode<E> node) {
         throw new UnsupportedOperationException();
     }
 
+    /**
+     * {@inheritDoc}
+     * <p>
+     * This method is only safe to call from the consumer thread!
+     */
+    @Override
+    public final boolean remove(Object o) {
+        if (null == o) {
+            return false; // Null elements are not permitted, so null will never be removed.
+        }
+
+        final LinkedQueueAtomicNode<E> originalConsumerNode = lpConsumerNode();
+        for (LinkedQueueAtomicNode<E> prevConsumerNode = originalConsumerNode, currConsumerNode = getNextConsumerNode(originalConsumerNode);
+             currConsumerNode != null;
+             prevConsumerNode = currConsumerNode, currConsumerNode = getNextConsumerNode(currConsumerNode)) {
+            if (o.equals(currConsumerNode.lpValue())) {
+                LinkedQueueAtomicNode<E> nextNode = getNextConsumerNode(currConsumerNode);
+                if (nextNode != null && prevConsumerNode != originalConsumerNode) {
+                    // We are removing an interior node.
+                    prevConsumerNode.soNext(nextNode);
+
+                    // Avoid GC nepotism because we are discarding the current node.
+                    currConsumerNode.soNext(null);
+                } else if (prevConsumerNode == originalConsumerNode) {
+                    // We are removing the consumer node.
+                    prevConsumerNode.soNext(null);
```

Comment:
in the `getSingleConsumerNodeValue` the `next` is set to itself instead of `null` ... is this necessary?